packages/toolpad-app/src/server/data.ts

```diff
@@ -231,7 +232,9 @@ export async function createApp(name: string, opts: CreateAppOptions = {}): Prom
   }
 
   if (await prismaClient.app.findUnique({ where: { name: appName } })) {
-    throw new Error(`An app named "${name}" already exists.`);
+    throw new Error(`An app named "${name}" already exists.`, {
+      cause: TOOLPAD_UNIQUE_CONSTRAINT_VIOLATION,
```

Comment:
* We should only use `Error` instances in `cause`.
* Add a `code` property you import from `./src/errorCodes.tsx`:
    ```tsx
    export const ERR_APP_EXISTS = `TOOLPAD_ERR_APP_EXISTS`
    ```
    See also https://github.com/mui/mui-toolpad/pull/1272#issuecomment-1312401592